_Classes/DeusEx/Classes/DeusExPlayer.uc

```diff
@@ -2399,6 +2397,80 @@ function RecoilEffectTick(float deltaTime)
 	}
 }
 
+// ----------------------------------------------------------------------
+// GetMeleePriority()
+// ----------------------------------------------------------------------
+
+function GetMeleePriority(int damageThreshold)	// Trash: Used to automatically decide what to draw
+{
+	local Inventory anItem;
+	local DeusExWeapon meleeWeapon;
+
+	local DeusExWeapon crowbar, sword, knife, baton, dts;
+
+	For(anItem = Inventory; anItem != None; anItem = anItem.Inventory)	// Go through the entire inventory, check for these melee weapons
+	{
+		if (anItem.IsA('WeaponSword'))
+			sword = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponCrowbar'))
+			crowbar = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponCombatKnife'))
+			knife = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponBaton'))
+			baton = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponNanoSword'))
+			dts = DeusExWeapon(anItem);
+	}
+
+
+	if (bHardcoreMode)
+	{
+		if (sword != none)	// Prioritize weapons better at breaking containers
+			meleeWeapon = sword;
+		else if (crowbar != none)
+			meleeWeapon = crowbar;
+		else if (knife != none)
+			meleeWeapon = knife;
+		else if (baton != none)
+			meleeWeapon = baton;
+		else if (dts != none)
+			meleeWeapon = dts;
+	}
+	else
+	{
+		if (sword != none && BreaksDamageThreshold(sword, damageThreshold))	// Prioritize weapons based on priority and container DT
+			meleeWeapon = sword;
+		else if (crowbar != none && BreaksDamageThreshold(crowbar, damageThreshold))
+			meleeWeapon = crowbar;
+		else if (knife != none && BreaksDamageThreshold(knife, damageThreshold))
+			meleeWeapon = knife;
+		else if (baton != none && BreaksDamageThreshold(baton, damageThreshold))
+			meleeWeapon = baton;
+		else if (dts != none && BreaksDamageThreshold(dts, damageThreshold))
+			meleeWeapon = dts;
+	}
+	
+	if (meleeWeapon != None)
+		PutInHand(meleeWeapon);
+}
+
+function bool BreaksDamageThreshold(DeusExWeapon weapon, int damageThreshold)	// Checks if the weapon breaks the damageThreshold
+{
+	local int mod;
+
+	if (weapon.IsA('WeaponCrowbar'))	// Special check for Crowbar since it deals +5 extra damage to objects
+		mod += 5;
+
+	if ((weapon.CalculateTrueDamage() + mod) >= damageThreshold)
```

Comment:
can replace this entire if with `return weapon.CalculateTrueDamage() + mod) >= damageThreshold;` to make the function a lot smaller.

Reply:
Gonna make a different PR called "Trash's Code Cleanup #1" that handles that, and fixes tiny bugs along with commenting every now and then.